java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java

```diff
@@ -58,12 +58,27 @@ public static int bitIndex(int absoluteBitIndex) {
   public static void setValidityBitToOne(ArrowBuf validityBuffer, int index) {
     final int byteIndex = byteIndex(index);
     final int bitIndex = bitIndex(index);
-    byte currentByte = validityBuffer.getByte(byteIndex);
-    final byte bitMask = (byte) (1L << bitIndex);
+    int currentByte = validityBuffer.getByte(byteIndex);
+    final int bitMask = 1 << bitIndex;
     currentByte |= bitMask;
     validityBuffer.setByte(byteIndex, currentByte);
   }
 
+  /**
+   * Set the bit at provided index to 0.
+   *
+   * @param validityBuffer validity buffer of the vector
+   * @param index index to be set
+   */
+  public static void setValidityBitToZero(ArrowBuf validityBuffer, int index) {
+    final int byteIndex = byteIndex(index);
```

Comment:
can this delegate to setValidityBit instead?  I think it is likely JIT will inline and do dead-code elimination?

Reply:
I am sorry I don't understand this comment. You mean this method should be called from setValidityBit?

Reply:
sorry I thought I replied here, the performance benchmarks show the JIT will not eliminate the branches.  we should comment on why code duplication is necessary instead of calling setValidityBit below.

Reply:
I see. Thanks for the clarification.
I have added comments explicitly to discuss this.